main.tf

```
@@ -353,7 +353,7 @@ resource "aws_autoscaling_group" "this" {
     for_each = var.use_mixed_instances_policy ? [] : [1]
 
     content {
-      name    = local.launch_template
+      id      = aws_launch_template.this[0].id
```

I think it is better to introduce a new local after line 8, which will use `id` attribute like this:
```
launch_template_id = var.create_launch_template ? aws_launch_template.this[0].id : var.launch_template_id
```

Add a new optional `var.launch_template_id`.

And have this in several places suggested by this PR already:
```suggestion
      id      = local.launch_template_id
```

I also have not run an example code with old and new code to verify that it doesn't break anything. Could you do that, @dhoppe ?

---

Also, I think a similar update should be implemented on line 507.

---

thanks for testing and validating @dhoppe - once you address the feedback provided above we should be all set to merge. thank you!

---

looking at this again (not on mobile this time!) I think the change is absolutely valid, but its now a breaking change

if users are passing in an externally created launch template today, they will have to change their code to pass in a launch template ID instead (and we need to fix this PR to support that functionality) 

this is the reason for the local variable - it helps to ensure we are using the correct value in multiple places by doing the conditional logic check once (to use an externally provided launch template ID or the ID of the launch template the module creates)

---

Yeah. I am already trying to get my head around `var.launch_template` vs `var.launch_template_name`. The naming of this variables is a bit confusing. 

The 3rd commit should allow to keep the functionality for already existing launch templates, but it is still a breaking change because I renamed the variable. 

To be honest, I have not tested the code yet because I need to take care of my 4 year old unless she wants to learn one or two things about AWS and Terraform. ;)